web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -219,7 +221,40 @@ public String getFullRecord(String pidString, Model model, HttpServletRequest re
         }
 
         if (briefObject.getResourceType().equals(searchSettings.resourceTypeAggregate)) {
-            model.addAttribute("viewerNeeded", accessCopiesService.hasViewableFiles(briefObject, principals));
+            boolean imageViewerNeeded = accessCopiesService.hasViewableFiles(briefObject, principals);
+            model.addAttribute("imageViewerNeeded", imageViewerNeeded);
+
+            boolean pdfViewerNeeded = false;
+
+            if (!imageViewerNeeded) {
+                // Check if primary object is a pdf
+                pdfViewerNeeded = accessCopiesService.hasViewablePdf(briefObject, principals);
+
+                // Check if first child is a pdf
+                if (!pdfViewerNeeded) {
+                    // Get child objects
+                    SearchRequest searchRequest = new SearchRequest();
+                    SearchState searchState = searchStateFactory.createSearchState();
+                    searchState.setFacetsToRetrieve(null);
+                    searchState.setIgnoreMaxRows(true);
```

Comment:
Can you replace this setting with setting the number of rows to 1? There's no need for any rows past the first one.